rsciio/delmic/_api.py

```diff
@@ -1,80 +1,804 @@
+
+    # Check for the expected dimensions
+    if Image.shape[0] < 1 or Image.shape[3] < 1 or Image.shape[4] < 1:
+        raise ValueError(
+            "The input 'Image' does not have the expected dimensions."
+        )
+
+    # Extract and transpose the data
+    data = Image[:, 0, 0, :, :].transpose(1, 2, 0)
+
+    return data
+
+
+def load_streak_camera_image(Image):
+    """Extract a a 4D streak camera image from a 5D dataset."""
+    # Check if Image is a h5py dataset
+    if not isinstance(Image, h5py._hl.dataset.Dataset):
+        raise TypeError("The input 'Image' must be a h5 dataset.")
+
+    # Check if Image has 5 dimensions
+    if Image.ndim != 5:
+        raise ValueError("The input 'Image' must be a 5D h5 dataset.")
+
+    # Check for the expected dimensions
+    if (
+        Image.shape[0] < 1
+        or Image.shape[1] < 1
+        or Image.shape[3] < 1
+        or Image.shape[4] < 1
+    ):
+        raise ValueError(
+            "The input 'Image' does not have the expected dimensions."
+        )
+
+    # Extract and transpose the data
+    data = Image[:, :, 0, :, :].transpose(2, 3, 1, 0)
+    # .transpose(3, 2, 0, 1)
+
+    return data
+
+
+def load_AR_spectrum(Image):
+    """Extract an Angle-Resolved Spectrum (aka E-k) from a 5D dataset."""
+    # Check if Image is a h5py dataset
+    if not isinstance(Image, h5py._hl.dataset.Dataset):
+        raise TypeError("The input 'Image' must be a h5 dataset.")
+
+    # Check if Image has 5 dimensions
+    if Image.ndim != 5:
+        raise ValueError("The input 'Image' must be a 5D h5 dataset.")
+
+    # Check for the expected dimensions
+    if (
+        Image.shape[0] < 1
+        or Image.shape[1] < 1
+        or Image.shape[3] < 1
+        or Image.shape[4] < 1
+    ):
+        raise ValueError(
+            "The input 'Image' does not have the expected dimensions."
+        )
+
+    # Extract and transpose the data
+    data = Image[:, :, 0, :, :].transpose(2, 3, 1, 0)
+
+    return data
+
+
+def load_temporal_trace(Image):
+    """Extract a decay trace or g(2) curve from a 5D dataset."""
+    # Check if Image is a h5py dataset
+    if not isinstance(Image, h5py._hl.dataset.Dataset):
+        raise TypeError("The input 'Image' must be a h5 dataset.")
+
+    # Check if Image has 5 dimensions
+    if Image.ndim != 5:
+        raise ValueError("The input 'Image' must be a 5D h5 dataset.")
+
+    # Check for the expected dimensions
+    if Image.shape[1] < 1 or Image.shape[3] < 1 or Image.shape[4] < 1:
+        raise ValueError(
+            "The input 'Image' does not have the expected dimensions."
+        )
+
+    # Extract and transpose the data
+    data = Image[0, :, 0, :, :].transpose(1, 2, 0)
+
+    return data
+
+
+def get_unit_prefix(number):
+    """Return the SI prefix for the given number based on its magnitude."""
+    if 1e-15 < np.abs(number) < 1e-12:
+        prefix = "f"
+    if 1e-12 < np.abs(number) < 1e-9:

```

Comment:
Based on the above CodeQL warning, shouldn't this be an `elif` as well?
Always one end of the range needs to be `<=`, because otherwise `1e-12, 1e-9, 1e-6, 1e-3, 1` are passed to the final `else`.
However, there might be a more elegant way to handle this using `pint`?